### ISSUE TITLE
[texlive-luatex] zlib library version does not match - header: 1.2.13, library: 1.3

### ISSUE THREAD

**Issue by That-Human-Being** [msys2/MINGW-packages#18251]
### Description / Steps to reproduce the issue

Run a command like `fmtutil`, `updmap`, `mtxrun`, etc.
(These are the commands that Pacman runs after installing any texlive packages)

### Expected behavior

The commands run without error?
Quite literally my google searching suggested that `luatex` just needs to be rebuilt with the new zlib version

### Actual behavior

The packages install fine, it's just the post transaction hooks that fail.

```
:: Running post-transaction hooks...
(1/3) Updating TeXLive filename database...
>>> Running mktexlsr
>>> Running mtxrun (this might take a while)
PANIC: unprotected error in call to Lua API (zlib library version does not match - header: 1.2.13, library: 1.3)
/
(2/3) Updating TeXLive format files...
>>> Running fmtutil (this might take a while)
PANIC: unprotected error in call to Lua API (zlib library version does not match - header: 1.2.13, library: 1.3)
error: command (/usr/share/libalpm/scripts/mingw-w64-x86_64-texlive-fmtutil.script /usr/share/libalpm/scripts/mingw-w64-x86_64-texlive-fmtutil.script ) failed to execute correctly
(3/3) Updating TeXLive font maps...
>>> Running updmap (this might take a while)
PANIC: unprotected error in call to Lua API (zlib library version does not match - header: 1.2.13, library: 1.3)
error: command (/usr/share/libalpm/scripts/mingw-w64-x86_64-texlive-updmap.script /usr/share/libalpm/scripts/mingw-w64-x86_64-texlive-updmap.script ) failed to execute correctly
```

### Verification

- [X] I have verified that my MSYS2 is up-to-date before submitting the report (see https://www.msys2.org/docs/updating/)

### Windows Version

MINGW64_NT-10.0-19044

### MINGW environments affected

- [X] MINGW64
- [ ] MINGW32
- [x] UCRT64
- [x] CLANG64
- [ ] CLANG32
- [ ] CLANGARM64

**Comment by lazka** [comment]
thanks.

huh, https://github.com/TeX-Live/texlive-source/blob/0d226662f0ab16f2bd11d5ab4e1cdbf2b7407b79/texk/web2c/luatexdir/luazlib/lzlib.c#L551

I wonder what's the point of that, besides everything breaking.

**Comment by lazka** [comment]
fixed via #18253

thanks for the report

**Comment by asd-a** [comment]
> fixed via #18253
> 
> thanks for the report

the same bug reports again, like `zlib library version does not match - header: 1.3, library: 1.3.1`

for `texlive-luatex-2023.20230404-1`and  `zlib-1.3.1-1`

**Comment by lazka** [comment]
Thanks, see #19921
